src/codegen/replacements.ts

```diff
@@ -81,95 +128,103 @@ export const enums = {
   $streamErrored: "3",
   $streamReadable: "4",
   $streamWaiting: "5",
   $streamWritable: "6",
-
-  "process.platform": JSON.stringify(Bun.env.TARGET_PLATFORM ?? process.platform),
   "process.arch": JSON.stringify(Bun.env.TARGET_ARCH ?? process.arch),
+  "process.env.NODE_ENV": JSON.stringify(debug ? "development" : "production"),
+  "process.platform": JSON.stringify(Bun.env.TARGET_PLATFORM ?? process.platform),
 };
 
 // ------------------------------ //
 
 for (const name in enums) {
   const value = enums[name];
-  if (typeof value !== "object") throw new Error("Invalid enum object " + name + " defined in " + import.meta.file);
-  if (typeof value === null) throw new Error("Invalid enum object " + name + " defined in " + import.meta.file);
-  const keys = Array.isArray(value) ? value : Object.keys(value).filter(k => !k.match(/^[0-9]+$/));
-  define[`$${name}IdToLabel`] = "[" + keys.map(k => `"${k}"`).join(", ") + "]";
-  define[`$${name}LabelToId`] = "{" + keys.map(k => `"${k}": ${keys.indexOf(k)}`).join(", ") + "}";
+  if (typeof value !== "object" || value === null) {
+    throw new Error(`Invalid enum object ${name} defined in ${import.meta.file}`);
+  }
+  const keys = Array.isArray(value) ? value : Object.keys(value).filter(k => !/^\d+$/.test(k));
+  const idToLabel = `$${name}IdToLabel`;
+  const labelToId = `$${name}LabelToId`;
+  if (keys.length) {
+    define[idToLabel] = `["${keys.join('", "')}"]`;
+    define[labelToId] = `{${keys.map((k, i) => `"${k}": ${i}`).join(", ")}}`;
+  } else {
+    define[idToLabel] = "[]";
+    define[labelToId] = "{}";
+  }
 }
 
 for (const name of globalsToPrefix) {
-  define[name] = "__intrinsic__" + name;
+  define[name] = `__intrinsic__${name}`;
 }
 
 for (const key in define) {
   if (key.startsWith("$")) {
-    define["__intrinsic__" + key.slice(1)] = define[key];
+    define[`__intrinsic__${key.slice(1)}`] = define[key];
     delete define[key];
   }
 }
 
+type Replacer = (substring: string, ...args: any[]) => string;
 export interface ReplacementRule {
   from: RegExp;
-  to: string;
+  to: string | Replacer;
   global?: boolean;
 }
 
 /** Applies source code replacements as defined in `replacements` */
 export function applyReplacements(src: string, length: number) {
   let slice = src.slice(0, length);
-  let rest = src.slice(length);
-  slice = slice.replace(/([^a-zA-Z0-9_\$])\$([a-zA-Z0-9_]+\b)/gm, `$1__intrinsic__$2`);
+  // Replace symbol @@ names first.
+  for (const name of symbolPropsToPrefix) {
+    // Replace symbol snapshot property references.
+    slice = slice.replaceAll(`.$$${name}`, `.__intrinsic__$${name}`);
+    // Replace symbol live property references.
+    slice = slice.replaceAll(`[$$${name}]`, `.__intrinsic____intrinsic__${name}`);
+  }
+  slice = slice.replace(/([^\$\w])\$(\w+\b)/gm, "$1__intrinsic__$2");
   for (const replacement of replacements) {
-    slice = slice.replace(replacement.from, replacement.to.replaceAll("$", "__intrinsic__").replaceAll("%", "$"));
```

Comment:
Move `replacement.to` transforms higher to ensure they are performed once and to improve consistency of transformation between `replacements` and `globalReplacements`.